diff --git a/src/renderer/Renderer.cpp b/src/renderer/Renderer.cpp
index 78faec3..572702e 100644
--- a/src/renderer/Renderer.cpp
+++ b/src/renderer/Renderer.cpp
@@ -250,6 +250,9 @@ void CRenderer::renderRect(const CBox& box, const CColor& col, int rounding) {
 }
 
 void CRenderer::renderTexture(const CBox& box, const CTexture& tex, float a, int rounding, std::optional<wl_output_transform> tr) {
+    if (tex.m_iTarget == TEXTURE_INVALID)
+        return;
+
     float matrix[9];
     wlr_matrix_project_box(matrix, &box, tr.value_or(WL_OUTPUT_TRANSFORM_FLIPPED_180) /* ugh coordinate spaces */, 0,
                            projMatrix.data()); // TODO: write own, don't use WLR here
